include/envoy/http/header_map.h

```diff
@@ -352,6 +352,9 @@ class HeaderEntry {
   virtual const HeaderEntry* name() const PURE;                                                    \
   virtual HeaderEntry* name() PURE;                                                                \
   virtual HeaderEntry& insert##name() PURE;                                                        \
+  virtual void setReference##name(const std::string& value) PURE;                                  \
+  virtual void setCopy##name(absl::string_view value) PURE;                                        \
+  virtual void setInteger##name(uint64_t value) PURE;                                              \
```

Comment:
Do we really need the integer interface to all headers? We only set integer value to a limited set of headers like status, grpc-status, content-length. Since now this code is generated via macro I think we can generate less?

Reply:
Oh interesting. Yes there's < 10 headers that need this interface, we definitely don't need all of them. I will try working this out.